src/block_ffm.rs

```diff
@@ -375,24 +461,42 @@ impl<L: OptimizerTrait + 'static> BlockTrait for BlockFFM<L> {
                     let feature = fb.ffm_buffer.get_unchecked(ffm_buffer_index);
                     let feature_index = feature.hash as usize;
                     let feature_value = feature.value;
+                    let feature_value_simd = f32x4::splat(feature_value);
 
                     if feature_num == 0 {
                         // first feature of the field - just overwrite
-                        for z in 0..field_embedding_len_as_usize {
+                        for z in (0..field_embedding_len_end).step_by(STEP) {
+                            let ffm_weights_simd = f32x4::from_slice(ffm_weights.get_unchecked(feature_index + z..feature_index + z + STEP));
```

Comment:
f32x4 -> this is a general question similar to @yonatankarni 's I guess. One would imagine `k` and possibly num fields etc. could impact this choice's optimality - perhaps worth testing

Reply:
I tested with multiple SIMD sizes & didn't really see any gain while increasing or decreasing.